### DISCUSSION TITLE
Approaches and lessons learned

### DISCUSSION THREAD

**Discussion by ricardoevvargas** [petrobras/3W#29] [Ideas]
**Do you have an idea on how to explore the 3W's features and would you like to ask our community for opinions?**

This is our dedicated space for this type of interaction. For each idea, leave a comment for others to reply. It is hoped that this discussion will help community members achieve their goals more efficiently by re-using approaches and lessons learned by other members.

Thanks for the contribution!

**Comment by utkuaslan85** [comment]
First of all, I am pleased to thank you for this work. It is almost impossible to find publicly available dataset related with oil & gas for freelance researchers or students to study on. This work may be really helpful for some like me.
I am a MSc Subsea Engineering student at Robert Gordon University (Online-distance learning). And I am also an R&D Specialist on AI. I am planning to work on 3W dataset in my MSc dissertation which I will start this semester. I am thinking to use GANs to create synthetic data to increase data and check whether this will increase performance of ML algorithms I will try in the study.
Can you please share your kind opinion about my proposal? If you think this is not a good idea, I would be happy to hear your advice if you want to share.

**Comment by cjmunaro** [comment]
Your proposal is very attractive, since most anomalies 5,6,7,8 have less than 8 instances (times series) of real data. Care must be taken, however, due to the diversity of behaviors of data in different time series. I suggest to work at instance level ot at least grouping instances with similar behavior. As an example, I attach a figure showing 5 variables for anomaly number 8 for the three available instances. You can see the behavior of the variables can be very different for the same anomaly in different instances. Data was normalized and time-shifted to produce this figure.
image
![Anomaly_number_8](https://user-images.githubusercontent.com/112667538/191003858-1216db1e-0740-4371-9a73-2e88f9cdd2f2.jpg)

**Reply by utkuaslan85** [reply]
Thank you very much for the advice and the figure. I started to work in detail to find best approach to deal such data distribution.

**Reply by ricardoevvargas** [reply]
Hi, @utkuaslan85. Glad you liked the 3W and that it will be useful for your work. I agree with @cjmunaro: your proposal is really interesting. It is expected that some datasets (including the 3W dataset) have event types that are rare, critical and difficult to model with a phenomenological approach. In these scenarios, we generate simulated and hand-drawn instances. Verifying the hypothesis that a GAN is able to improve the training process of a ML algorithm to the point of increasing its performance with a test set is a very useful research. Depending on the result of this research, we will have one more approach to help in the detection/classification of real events.
